src/androidTest/java/com/couchbase/lite/support/BatcherTest.java

```diff
@@ -285,6 +282,14 @@ public void process(List<String> itemsToProcess) {
             success = latch1.await(500, TimeUnit.MILLISECONDS);
             assertTrue(success);
 
+            // NOTE: https://github.com/couchbase/couchbase-lite-java-core/issues/518
+            //       CountDownLatch became 0, but task might not be done yet.
+            //       Give little bit of time for slower environment.
+            try {
+                Thread.sleep(500);
+            } catch (Exception ex) {
+            }
+
             // add another object
             batcher.queueObject(new String());
```

Comment:
@pasin  Thank you very much for your feedback. Instead of changing BatcherTest, I will try to fix Batcher itself. Thanks!